### PR TITLE
Fix BUCKET cannot accept column name and negative boundaries bug

### PR DESCRIPTION
**Pull request by sneaxiy**
Add `BUCKET` column support of:

- `BUCKET(col_name)`, where `col_name` is the string name of the column. Only `BUCKET(NUMERIC(col_name))` is supported before this PR.
- `BUCKET(col_name, [-10])`, where the boundaries contain negative number.

### CODE REVIEW

File: pkg/sql/ir_generator.go [hunk 1]
```diff
@@ -538,27 +564,43 @@ func parseNumericColumn(el *parser.ExprList) (*ir.NumericColumn, error) {
...
-	if err != nil {
-		return nil, err
-	}
-	if _, ok := source.(*ir.NumericColumn); !ok {
-		return nil, fmt.Errorf("key of BUCKET must be NUMERIC, which is %s", source)
+
+	var source ir.FeatureColumn
+	var err error
+
+	if sourceExprList.Sexp == nil {
```
Comment (typhoonzero) [inline pkg/sql/ir_generator.go:578]: Better to use `sourceExprList.Type` to determine whether the subexpression is a string or an expression.

File: pkg/sql/ir_generator.go [hunk 2]
```diff
@@ -440,13 +441,23 @@ func parseExpression(e interface{}) (interface{}, error) {
...
 					}
-				} else {
-					value, err := parseExpression(&expr.Sexp)
-					if err != nil {
-						return nil, err
+					continue
+				}
+
+				// parse negative integer
+				if len(expr.Sexp) == 2 && (*expr.Sexp[0]).Value == negative {
```
Comment (typhoonzero) [inline pkg/sql/ir_generator.go:448]: Can add a comment to reference extended_syntax_parser.y:368 for further reading.

**Review by typhoonzero — APPROVED** [review]
Just some minor comments.

File: pkg/sql/ir_generator.go (hunk 1 above)
Comment (sneaxiy) [reply pkg/sql/ir_generator.go:578]: OK. I will fix it in next PR asap.

File: pkg/sql/ir_generator.go (hunk 2 above)
Comment (sneaxiy) [reply pkg/sql/ir_generator.go:448]: Same above.

File: pkg/sql/ir_generator.go (hunk 1 above)
Comment (sneaxiy) [reply pkg/sql/ir_generator.go:578]: OK. I will fix it in next PR asap.

**Pull request merged**